### PR TITLE
Fixes the PR commit hash we publish npm modules under

### PR DESCRIPTION
**Pull request by foot**
- We derive the name of the npm artifact from the git ref.
- on.pull_request creates an intermediate merge commit and messes this up

**What changed?**

Updates the actions/checkout step to avoid a quirk of `on.pull_request` checking out the code to a "non-existant" merge commit. 

**Why was this change made?**

We use the ref from the commit to name the npm package. This should reflect the actual commit and not the temporary merge commit actions/checkout creates.

**How was this change implemented?**

Straight up yaml engineering.

**How did you validate the change?**

Checked the _publish js library_ CI output which now shows:
> `npm notice 📦  @weaveworks/weave-gitops-main@0.12.0-20-g116fd4bc`

For the head of this PR as of writing which matches the last commit.

**Release notes**

**Documentation Changes**

### CODE REVIEW

**Comment by foot** [comment]
@opudrovs you should be able to run `make update-weave-gitops BRANCH=fixes-publishing-npm-module-on-pr` on this branch!

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -249,6 +249,11 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
         with:
+          # avoid the merge commit that on.pull_request creates
+          # fallback to github.sha if not present (e.g. on.push(main))
+          # https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit
+          # We want the correct sha so we can tag the npm package correctly
+          ref: ${{ github.event.pull_request.head.sha || github.sha }}
```
Comment (foot) [inline .github/workflows/pr.yaml:256]: I don't know if this will work for main. I _think_ so. Other solutions?

**Comment by opudrovs** [comment]
@foot thanks, will have a look at it right after the sync.

**Review by makkes — APPROVED** [review]
I think it's worth a try. Thanks for addressing the issue, @foot.

**Comment by opudrovs** [comment]
Done with my PR, will check it now.

**Review by opudrovs — APPROVED** [review]
LGTM.

The tag appeared for me after running the command you provided (with enterprise main checked out). I hope it is working as expected:

**Pull request merged**